velox/parse/tests/QueryPlannerTest.cpp

```diff
@@ -0,0 +1,165 @@
+        nullptr, // nulls
+        0,
+        std::vector<VectorPtr>{});
+  }
+
+  std::unique_ptr<memory::MemoryPool> pool_{
+      memory::getDefaultScopedMemoryPool()};
+};
+
+TEST_F(QueryPlannerTest, values) {
+  assertPlan(
+      "SELECT x, x + 5 FROM UNNEST([1, 2, 3]) as t(x)",
+      "-- Project\n"
+      "  -- Unnest\n"
+      "    -- Project\n"
+      "      -- Values\n");
+
+  assertPlan(
+      "SELECT sum(x) FROM UNNEST([1, 2, 3]) as t(x)",
+      "-- Project\n"
+      "  -- Aggregation\n"
+      "    -- Unnest\n"
+      "      -- Project\n"
+      "        -- Values\n");
+
+  assertPlan(
+      "SELECT x % 5, sum(x) FROM UNNEST([1, 2, 3]) as t(x) GROUP BY 1",
+      "-- Project\n"
+      "  -- Aggregation\n"
+      "    -- Project\n"
+      "      -- Unnest\n"
+      "        -- Project\n"
+      "          -- Values\n");
+
+  assertPlan(
+      "SELECT sum(x * 4) FROM UNNEST([1, 2, 3]) as t(x)",
+      "-- Project\n"
+      "  -- Aggregation\n"
+      "    -- Project\n"
+      "      -- Unnest\n"
+      "        -- Project\n"
+      "          -- Values\n");
+}
+
+TEST_F(QueryPlannerTest, tableScan) {
+  std::unordered_map<std::string, std::vector<RowVectorPtr>> inMemoryTables = {
+      {"t",
+       {makeEmptyRowVector(
+           ROW({"a", "b", "c"}, {BIGINT(), INTEGER(), SMALLINT()}))}},
+      {"u", {makeEmptyRowVector(ROW({"a", "b"}, {BIGINT(), DOUBLE()}))}},
+  };
+
+  assertPlan(
+      "SELECT a, sum(b) FROM t WHERE c > 5 GROUP BY 1",
+      inMemoryTables,
+      "-- Project\n"
+      "  -- Aggregation\n"
+      "    -- Filter\n"
+      "      -- Values\n");
+
+  assertPlan(
+      "SELECT t.a, t.b, t.c, u.b FROM t, u WHERE t.a = u.a",
+      inMemoryTables,
+      "-- Project\n"
+      "  -- Filter\n"
+      "    -- CrossJoin\n"
+      "      -- Values\n"
+      "      -- Values\n");
+}
+
+TEST_F(QueryPlannerTest, customScalarFunctions) {
+  DuckDbQueryPlanner planner(pool_.get());
+  planner.registerScalarFunction("foo", {BIGINT()}, BOOLEAN());
+  planner.registerScalarFunction("bar", {ARRAY(BIGINT())}, BIGINT());
+
+  auto plan =
+      planner.plan("SELECT foo(x), bar([x]) FROM UNNEST([1, 2, 3]) as t(x)");
+  ASSERT_EQ(
+      plan->toString(false, true),
+      "-- Project\n"
+      "  -- Unnest\n"
+      "    -- Project\n"
+      "      -- Values\n");
+}
+
+TEST_F(QueryPlannerTest, customAggregateFunctions) {
+  DuckDbQueryPlanner planner(pool_.get());
+
+  planner.registerAggregateFunction("foo_agg", {BIGINT(), BIGINT()}, DOUBLE());
+  planner.registerAggregateFunction(
+      "bar_agg", {ARRAY(BIGINT()), BIGINT()}, ARRAY(BIGINT()));
+
+  auto plan = planner.plan(
+      "SELECT foo_agg(x, x + 5), bar_agg([x], 1) FROM UNNEST([1, 2, 3]) as t(x)");
+  ASSERT_EQ(
+      plan->toString(false, true),
+      "-- Project\n"
+      "  -- Aggregation\n"
+      "    -- Project\n"
+      "      -- Unnest\n"
+      "        -- Project\n"
+      "          -- Values\n");
+}
+
+TEST_F(QueryPlannerTest, error) {
```

Comment:
Maybe error on test of an aggregate function also.

Reply:
Adding

```
  // DuckDB gives the same error regardless of whether scalar or aggregate
  // function is missing.
  assertPlanError(
      "SELECT x % 5, my_agg(x) FROM UNNEST([1, 2, 3]) as t(x) GROUP BY 1",
      "Scalar Function with name my_agg does not exist");
```